python/pyspark/sql/pandas/_typing/__init__.pyi

```diff
@@ -347,10 +348,16 @@ PandasCogroupedMapFunction = Union[
 ArrowGroupedMapFunction = Union[
     Callable[[pyarrow.Table], pyarrow.Table],
     Callable[[Tuple[pyarrow.Scalar, ...], pyarrow.Table], pyarrow.Table],
+    Callable[[pyarrow.Table], Iterator[pyarrow.RecordBatch]],
+    Callable[[Tuple[pyarrow.Scalar, ...], pyarrow.Table], Iterator[pyarrow.RecordBatch]],
```

Comment:
Shouldn't we have iterator -> iterator instead of table -> iterator?

Reply:
+1, I also prefer `iterator -> iterator`